File: Easy/ValidParentheses.js
"Link: https://leetcode.com/problems/valid-parentheses/submissions/"
/**
 * @param {string} s
 * @return {boolean}
 */
 var isValid = function(s) {
    const start = ["(", "[", "{"];
    const endOf = {
        "}": "{",
        "]": "[",
        ")": "("
    };
    const stack = [];
    for (let i of s) {
        if (start.includes(i)) {
            stack.push(i);
        } else {
            const item = stack[stack.length-1];
            stack.pop();
            if (endOf[i] != item) {
                return false;
            }
        }
    }
    if (stack.length > 0) return false;
    return true;
};

File: Valid-Parentheses/Valid-Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack&lt;char&gt;st;
        for(int i=0;i&lt;s.length();i++){
            if(st.empty())
                st.push(s[i]);
            else if(s[i]==')'){
                if(st.top()!='(')return false;
                else st.pop();
            }
            else if(s[i]=='}'){
                if(st.top()!='{')return false;
                else st.pop();
            }
            else if(s[i]==']'){
                if(st.top()!='[')return false;
                else st.pop();
            }
            else
                st.push(s[i]);
        }
        if(st.empty())return true;
        return false;        
    }
};